docs/frameworks/mastra.mdx

```diff
@@ -0,0 +1,318 @@
+const add = tool({
+  parameters: z.object({
+    a: z.number(),
+    b: z.number(),
+  }),
+  execute: async (args) => {
+    return args.a + args.b;
+  },
+});
+
+// Multiply requires approval
+const multiplyTool = tool({
+  parameters: z.object({
+    a: z.number(),
+    b: z.number(),
+  }),
+  execute: async (args) => {
+    return args.a * args.b;
+  },
+});
+
+// Wrap multiply with approval requirement
+const multiply = hl.requireApproval({ multiplyTool });
+
+// Human consultation tool
+const contactHuman = hl.humanAsTool();
+
+// Create a Mastra agent
+const mathAgent = new Agent({
+  name: "math-agent",
+  instructions:
+    "You are a helpful math assistant. Use the provided tools to perform calculations.",
+  model: google("gemini-2.0-flash-001"),
+  tools: { add, multiply, contactHuman },
+  memory: new Memory(),
+});
+```
+
+## Chef Agent Example
+
+### Agent Definition
+
+```typescript src/mastra/agents/chefAgent.ts
+import { google } from "@ai-sdk/google";
+import { Agent } from "@mastra/core/agent";
+import { Memory } from "@mastra/memory";
+import { checkIngredients } from "../tools/kitchenTools";
+
+export const chefAgent = new Agent({
+  name: "chef-agent",
+  instructions:
+    "You are Michel, a practical and experienced home chef. " +
+    "You help people cook with whatever ingredients they have available. " +
+    "Use the checkIngredients tool to check what ingredients are available.",
+  model: google("gemini-2.0-flash-001"),
+  tools: { checkIngredients },
+  memory: new Memory(),
+});
+```
+
+### Tools Definition
+
+<Tip>
+  The tools below follow the Vercel AI SDK tool format for compatibility with
+  HumanLayer.
+</Tip>
+
+```typescript src/mastra/tools/kitchenTools.ts
+import { z } from "zod";
+import { tool } from "ai";
+import { humanlayer } from "humanlayer-vercel-ai-sdk";
+
+const hl = humanlayer({ verbose: true });
+
+const checkAvailableIngredients = tool({
+  description: "Checks the available ingredients",
+  parameters: z.object({}),
+  execute: async () => {
+    // Hardcoded list of available ingredients with categories and quantities
+    return [
+      {
+        category: "Proteins",
+        items: [
+          { name: "Chicken breast", quantity: "1 lb", expiresIn: "2 days" },
+          { name: "Ground beef", quantity: "0.5 lb", expiresIn: "1 day" },
+          { name: "Eggs", quantity: "6", expiresIn: "1 week" },
+          { name: "Tofu", quantity: "1 block", expiresIn: "5 days" },
+        ],
+      },
+      {
+        category: "Vegetables",
+        items: [
+          { name: "Onions", quantity: "3", expiresIn: "2 weeks" },
+          { name: "Bell peppers", quantity: "2", expiresIn: "4 days" },
+          { name: "Carrots", quantity: "5", expiresIn: "10 days" },
+          { name: "Spinach", quantity: "1 bag", expiresIn: "3 days" },
+          { name: "Tomatoes", quantity: "4", expiresIn: "5 days" },
+        ],
+      },
+      // More categories...
+    ];
+  },
+});
```

Comment:
canonical use cases with humanlayer tend to have no-approval on read operations, and only require approval on write operations...how about

```
{
    tools: [
      checkAvailableIngredients,
      hl.requireApproval(cook),
    ]
}
```